index.js

```diff
@@ -11,131 +11,107 @@ const fs = require('fs');
+                }
             }
         }
-    }
 
-    return app.mergeDeep(target, ...sources);
-};
+        return this.mergeDeep(target, ...sources);
+    };
+
+    // end *** https://stackoverflow.com/questions/27936772/how-to-deep-merge-instead-of-shallow-merge ******
+
+    /**
+     * Main recursive function to process swagger json
+     * @param object
+     * @returns {*}
+     */
+    traverse(object) {
+
+        if(typeof object !== 'object') return object;
+
+        if (Array.isArray(object)) return this.processArray(object);
+
 
-// end *** https://stackoverflow.com/questions/27936772/how-to-deep-merge-instead-of-shallow-merge ******
+        if (typeof object['$ref'] !== "undefined") return this.processRef(object);
 
-/**
- * Main recursive function to process swagger json
- * @param object
- * @returns {*}
- */
-app.traverse = function (object) {
-    if (typeof object === 'object') {
-        if (object.constructor === Array) {
-            return app.processArray(object)
-        } else {
 
-            if (typeof object['$ref'] !== "undefined") {
-                return app.processRef(object);
+        if (typeof object['allOf'] !== "undefined") return object['allOf'].reduce((acc, cur) => this.mergeDeep(acc, this.traverse(cur)), {});
+
+
+        return Object.entries(object).reduce((acc, cur) => {
+            const item = cur[1];
+            const key = cur[0];
+
+            if (typeof item === 'object') {
+                return Object.assign(acc, {[key]: Array.isArray(item) ? this.processArray(item) : this.traverse(item)});
             } else {
-                if (typeof object['allOf'] !== "undefined") {
-                    let merged = {};
-                    for (let i = 0; i < object['allOf'].length; i++) {
-                        let item = object['allOf'][i];
-                        let processedItem = app.traverse(item);
-                        merged = app.mergeDeep(merged, processedItem);
-                    }
-
-                    return merged;
-                } else {
-                    let toRet = {};
-                    for (let key in object) {
-                        let item = object[key];
-
-                        if (typeof item === 'object') {
-                            if (item.constructor === Array) {
-                                toRet[key] = app.processArray(item);
-                            } else {
-                                if (typeof item['$ref'] === "undefined") {
-                                    toRet[key] = app.traverse(item);
-                                } else {
-                                    toRet[key] = app.traverse(item);
-                                }
-                            }
-                        } else {
-                            toRet[key] = item;
-                        }
-
-                    }
-                    return toRet;
-                }
+                return Object.assign(acc, {[key]: item});
```

Comment:
можна робити так
```suggestion
                return { ..acc, {[key]: item} };
```

але то вже, як кому подобається, правда так не буде змінюватись вихідний об'єкт `acc`, тут це не важливо, але в цілому буде плюсом